### ISSUE TITLE
Unique XFCE "Theme" for Qubes to ship OEM with

### ISSUE THREAD

**Issue by ninavizz** [QubesOS/qubes-issues#6414]
**The problem you're addressing (if any)**
Qubes is a unique only-of-its-kind computing experience for its users. 

As such, windows designed to support GUI experiences within a non-containerized OS, seem to all meet about 80% of the needs for Qubes users... but then really miss the mark for the other 20% of our needs.

The resulting usability-salad kind of stinks, so a design to meet the needs of QubesOS users feels like an important thing to build to add to the usability of the QubesOS experience.

**Describe the solution you'd like**
Below created in a design-tantrum and is just a first stab. Some of the points are settings that can be modified for existing themes, but most have to do with the theme itself.
![Many Colors](https://user-images.githubusercontent.com/8262612/108578742-7b2d4980-72d8-11eb-8f10-b19976b8ee97.png)

- Simpler focus/background window states
	- Heavier dropshadow when in focus to make which window is on top clearer
	- Alpha on window title changes, but bar-background/color does not
- Qube icon to indicate type/color, in window’s upper-left corner
- Cleaner window control icons & fewer window icons by default
- Slightly taller top-bar to showcase window color
- “decoration” of top-bar limited to what can help its affordance of “grabbability” for users
- Default left-alignment of window title (easier to manage multiple windows with)
- Window title legibility w/ different colors mitigated with large/low-alpha dropshadow 
   (seen above) that matches  qube’s “shadow” hex. All dom0 and Template Qubes, wd have black.

**Where is the value to a user, and who might that user be?**
1. Good design always has value! :)
2. Legibility of window titles today, is complicated by the focus/background states in most of XFCE's window themes. Likewise, the gradient in many of the window bars feels either backwards, or detracts too much from the window's color identifying which qube a pane is from. 
3. Color as the lone disambiguator of windows as belonging to different qubes, has been cited as problematic. Including a qube's icon could also extend communicating the qube-type to a user, which could be nice.

**Describe alternatives you've considered**
I really have looked at every dang theme I could find, and they all come-up short in not addressing all of the various pain-points identified above.

**Additional context**
This did come about through a design tantrum, so I am entirely open to other ideas!

**Related, [non-duplicate](https://www.qubes-os.org/doc/reporting-bugs/#new-issues-should-not-be-duplicates-of-existing-issues) issues**
#5800

**Comment by DemiMarie** [comment]
Nit: Not every TemplateVM will have the “black” label.  It is quite possible to have a TemplateVM that isn’t fully trusted.  I have had a “red” TemplateVM in the past.

**Comment by ninavizz** [comment]
@DemiMarie Of course! The above is just a "proof-of-concept" mockup; eg, a halfass deliverable from a designer to communicate a general idea, not a complete spec.

**Comment by DemiMarie** [comment]
@ninavizz Should we deprecate the “gray” label?  It looks a bit too much like dom0, IMO.

**Comment by andrewdavidwong** [comment]
> @ninavizz Should we deprecate the “gray” label? It looks a bit too much like dom0, IMO.

This has come up before in a couple other issues: https://github.com/QubesOS/qubes-issues/issues/3471, https://github.com/QubesOS/qubes-issues/issues/3102.

**Comment by ninavizz** [comment]
@DemiMarie @andrewdavidwong My own $0.02 is that "system" applet-control windows should always display with a gray chrome, and dom0 Terminal and app windows with a Black chrome. 

I personally don't like that any app-VM may also be chosen to have a Black coloring of them, as I feel Black should be reserved for dom0, but I also want to ask about color stuff in my first round of user research. Templates and Service VMs should be allowed to be Black, but not App VMs. Again, my own, lay and not very engaged in the topic $0.02.

There are a few additional hues/colors that we don't include in the palette, that we could extend the pallete to include; if we'd want to remove gray and black as user-selectable options

I appreciate that many of the Qubes developer type users will identify something like an XFCE Settings applet and know which VM it's from, and expect that window to match that VM... but I also want to be cogent to map to more common mental models that both non-technical and technical users alike, can get their heads around. 

All of the above said: I DON'T want this ticket to digress into a discussion of near-term color stuff. If in fact that should be considered for 4.2 or later, I'd be happy to open a new/linked ticket to discuss it... but the mentioned things up above about the existing XFCE theme options, I feel are important to keep this ticket focused on. TL;DR, it'd be great to create a Qubes-specific theme, for all the reasons cited above.

**Comment by ninavizz** [comment]
@DemiMarie Also—the gray cute-qube, I did make to look much more like its own thing, and distinctly different from the black dom0 cute-qube. Is that what you're referring to, or the padlock?

Many of the existing colors are far too close in contrast to one another, and in the cute-qubes I did change the hues/contrast a bit to make them more clearly distinct. In the padlocks they're all somewhat muted, which is really bad for accessibility. So they got punched-up to be more contrast-y in the cute-qubes.

**Comment by andrewdavidwong** [comment]
https://github.com/QubesOS/qubes-issues/issues/2523

**Comment by hackerncoder** [comment]
> I personally don't like that any app-VM may also be chosen to have a Black coloring of them, as I feel Black should be reserved for dom0, but I also want to ask about color stuff in my first round of user research. Templates and Service VMs should be allowed to be Black, but not App VMs. Again, my own, lay and not very engaged in the topic $0.02.

I use it. Vault and master-gpg is black, and split-vault (used for gpg) and storage is just white, this signals that vault and master-gpg have *very* valuable stuff inside that I do not want out of them at any time, whereas split-vault and storage are non-networked VMs, and for one of them it would be nice if the contents didn't get out (but not critical). If black got removed from AppVMs I would need to figure out something else, and while I can (probably just use purple, which I don't use currently..) the current setup just feels right™.

Of course, this is an entire study one could do on what users use the colours for, and what they signal. Maybe such a thing should be done?

**Comment by brendanhoar** [comment]
A study would be useful.

I set all my templates to black, and all my sys- VMs (quite a few due to VPNs) to gray.

‘qvm-prefs dom0’ says label is black, but window title bar always appears silver/white (via gradient), which is fine.

For task VMs, the rest of the color palette is used as work context or security context dependent color signifiers.

E.g. All non-networked VMs (such as vault) are blue (I also re-verify that truth from time to time!).

Brendan

**Comment by hackerncoder** [comment]
I think I should probably create an issue for this, but I setup limesurvey to create a study/survey.

**Comment by th5** [comment]
That looks really nice. This would be a welcome improvement.

**Comment by ninavizz** [comment]
@brendanhoar So... to answer "what do users _want_," a study is a fantastic idea. Likewise, to gain insight into how users of qubes typically use their desktops—and from there, to prioritize design needs.

Unfortunately the default XFCE utilities Qubes uses, were all designed for single-OS environments. As such, they're designed to fulfill usability needs that befit a different mental-model than those Qubes users need to work within. Likewise, very little of any of what XFCE utilities were built for, follow any known best practices for either visual or interaction design interests.

Some really basic and small things can be done w/o a study, to create a theme that would better serve Qubes users' needs—by just adhering to many best practices for cognitive needs; both around basic/known design problems, such as text legibility with multiple window colors _and_ focus/non-focus windowbar states, and text legibility. 

Color is not in my mind the core problem. Rather, how the UI of the windows themselves, are rendered. The above proposal does a lot of very subtle "skinning" level things, to enhance legibility and quicker recognition of differently colored windows. It also has a qube icon in it, which would just be nice. None of those things are "necessary" when all windows are going to be the same color, or a user is unlikely to have a zillion windows open at once with 3 essential pieces of information on each title bar (vm, app, filename) vs the usual 2 (app, filename). 

@hackerncoder I'd love to see both your survey, and what you learn from it!

**Comment by ninavizz** [comment]
TL;DR: the universal "skinning" needs of one XFCE theme, with the existing colorset for Qubes, is the objective of this issue and the development need I'm flagging here. 

The colors vs textures vs "is color the right way to distinguish qubes?" I see as a far bigger ux inquiry, and to keep things "clean" unrelated to this. Both feel really important; as one could make a lot of peoples' lives easier in the near-term, while the other is important longer-term and is a far bigger "problem" to solve for.

**Comment by hackerncoder** [comment]
@ninavizz Here is my survey: https://survey.encryptionin.space/index.php?r=survey/index&sid=178339&lang=en. I would love some feedback (I'm a developer, not a UX person) before I send it out.

**Comment by ninavizz** [comment]
Thank you @hackerncoder for creating that survey! To keep this Issue focused on the creation of a new XFCE theme for the reasons articulated above, I've created [a separate issue](https://github.com/QubesOS/qubes-issues/issues/6463#issuecomment-798988959) for your survey—and included my responses, there. :)

**Comment by ffsec** [comment]
There is no need to ban or restrict any color at all.
It doesn't make much sense to reserve the black label for dom0. If you switch to the right red colored theme, dom0 windows will look like red label vms. It all comes down to the theme you are using.

As I have already written in https://github.com/QubesOS/qubes-issues/issues/2523#issuecomment-842786543, I have been working on this topic about 2 years ago. When I came up with the initial colors (https://github.com/QubesOS/qubes-issues/issues/2523#issuecomment-458219172), I had the current tinting mode of Qubes OS in mind. I ditched that idea shortly after, since it's just not good. So I moved on with true colors (no tinting), which allows the user to basically use any color (for example via color picker). I've also created some basic themes, one is Qubes-Dark with 5 different security levels (similar to Defcon). The user can choose how much coloring he/she wants. I've been using SL5 for quite some time and I love it. It gives me all the color/vm information I need in the top left corner (label color + vm name), while maintaining a uniform look and feel throughout the whole system. But if you prefer more coloring, you can also chose that.
Here are some screenshots I took from my dev machine (the automatic contrast window title text color is not patched in this version):

SL5 (lowest security level - my current theme):
![Qubes_Theme_SL5](https://user-images.githubusercontent.com/31329006/120082819-97e5d400-c0b4-11eb-8f2e-3315e6faef67.png)

SL4:
![Qubes_Theme_SL4](https://user-images.githubusercontent.com/31329006/120082830-a8964a00-c0b4-11eb-98e4-b38f01bf690b.png)

SL3:
![Qubes_Theme_SL3](https://user-images.githubusercontent.com/31329006/120082835-adf39480-c0b4-11eb-874e-ed792c936000.png)

SL2:
![Qubes_Theme_SL2](https://user-images.githubusercontent.com/31329006/120082841-b21fb200-c0b4-11eb-8436-71068f77337e.png)

SL1 (highest security level):
![Qubes_Theme_SL1](https://user-images.githubusercontent.com/31329006/120082847-bc41b080-c0b4-11eb-889b-86a07956c1ca.png)

I haven't finished them all to the last detail, for example in SL1-4 the dom0 windows look similar to black vm windows (which is actually just a darker grey as label color, beacuse it looks better with the button shadows, but you can also use real black if you really want to). This can can be easily fixed though.
As you can see in SL5 (which I'm using), dom0 windows have the blue Qubes logo (or grey Qubes logo for inactive windows), while all vm windows have the label color as icon. This is a unique look only for dom0, which is impossible for vms to mimic. Right next to that area is the vm name. I personally love dark themes, as well as a uniform look, so SL5 is just perfect for me.

As you can see, if you switch from tinting to using real colors, you can have any color you want, without restrictions.
Here are some more examples, all real windows from my dev machine, using the SL1 theme:
![Qubes_Theme_SL1_All_colors_possible](https://user-images.githubusercontent.com/31329006/120083634-83580a80-c0b9-11eb-8790-9693d398d6e6.png)

Regarding current (old) themes: Most of the old themes that come with Qubes OS can still be used. Some are working out of the box, some need some automatic fixing (I've done that) and some would need some manual fixing in order to make the buttons look better (they work, it's just an optical matter). Very few are incompatible.
Since none of the existing xfce theme was built with Qubes OS in mind (except the ones I showed you above), it's imo a good idea to create some real Qubes OS themes. They will be created specially for Qubes and different window colors and will look more modern and professional.
I went through all oft them 2 years ago and there was no standards applied for creating them. Every way was ok, as long as the result looked as intended. They are a mess. That's why I think it's even more important to have real Qubes OS themes. (People can still create or port their own themes if they want.)

In addition, the tinted application icons need to go as well, as they are generated using the same inferior tinting technique.
Using the real icon and adding a smaller label-colored badge might be one possibility.
With a new Qubes OS start menu, we probably wouldn't even need colored icons there anyway, since it would be clear from the menu itself, where these applications belong to. So it would all come down to desktop icons and tray icons? I'm pretty sure we can find a good solution for these, too.

Alright, I'm gonna stop here, as I've already written way more than i was going to when I started. :)

**Comment by ninavizz** [comment]
@ffsec Thank you for your ongoing contribution to the colors topic! For better and for worse (for the better of the _project_, for the worse of our _individual habits_ and inclination to want to contribute anywhere/anyhow) @andrewdavidwong is working on tightening-up how all of us contribute to issues, here. Namely—keeping more disciplined about staying on-topic, as well as improving how actionable contributions to issues are. Design is tough, as it requires a lot of conversation. 

Might you be keen to re-place your comment in #2523? Also—yes, i am very with you on the inferiority of coloring app icons. I've incorporated that into solutions proposed in addressing #5677 and #6366.

FWIW, I am possibly one of the greatest "offenders" of introducing segues and chaotic off-topic contributions—so pls know I make this request in the friendliest and most comrade-ish spirit of collaboration.  :)

**Comment by deeplow** [comment]
@ninavizz, @unman just shared [on another thread](https://qubes-os.discourse.group/t/windows-camouflage-for-qubes/4759/7) a windows-looking theme whose windows seem to be a close match to what you desire. And the most important detail of all is they support colors. Find instructions for using this theme on Qubes OS [here](https://qubes-os.discourse.group/t/windows-camouflage-for-qubes/4759/10).

![da593192fb5f33e15b313cd8483a8727da3d1d97_2_666x500](https://user-images.githubusercontent.com/47065258/123953418-ba5e5c00-d996-11eb-80ed-1e1df34a381b.png)

Perhaps this could be a good base to develop proposed theme on?

**Comment by ninavizz** [comment]
@deeplow How things are coded, is not my wheelhouse; but I'm extremely hesitant to "base" a theme off of anything, because of Qubes' unique needs wrt colorization, text legibility in focus and not, and the need to have a very basic, un-adorned solid color frame around all 4 edges of a window. No existing themes do all of that, or even most of that. 

My proposed PNG above, specifically also handles focus/not with the alpha value on both the text in windows and the super subtle top-bar gradient. Most window themes do focus in a different way, or don't do good dropshadows to provide clear layering of windows, or do no frame on 3 sides (as the one you shared does). 

I appreciate you sharing @unman's solution! This ticket is is really an issue to "SOLVE" a problem, instead of our usual... which is to devise more workarounds to "make better" something that is problematic, today. If that makes sense. :)

**Comment by deeplow** [comment]
@ninavizz I totally get what you are saying. It's just that implementation-wise any theme customization will likely be best based on a pre-existing theme, I can image. And the current theme (visually-speaking) is further apart from the one you propose.

**Comment by ninavizz** [comment]
@deeplow Yup! I'd rather leave those decisions to whomever wishes to tackle the development of a solution to this issue, tbh. That individual will know which foundational pieces they want to retain for a pre-existing template to work from, vs which require more work to customize. 

The current theme has borders on all 4 sides of the window, as an example. That could make it easier to customize, than the theme you propose in this issue (which has no borders on 3 sides)—because the developer would just need to specify the same color for the outer stroke, the inner-stroke, and the frame color, while also eliminating the gradient shading. In that regard, I feel the static screen example you provide is much farther away from the end goal, than the current theme.

Let's just leave this issue where it's at, until a developer is ready to pick it up to tackle it—and then they can drive what they need?

**Comment by ninavizz** [comment]
Labeled this issue "Help Wanted," because none of the core Qubes OS team are likely to be able to prioritize this anytime soon, abreast critical project priorities. That said: to me, a custom XFCE window theme feels like an helpful addition to the Qubes OS UX. 

The core Qubes team have given their blessing to this UX solution, so should a developer like to take this on pls "@" me here and let's chat next steps. I will gladly make myself available to support that work, once someone decides to take this on.

**Comment by arkenoi** [comment]
Speaking of window decorations, I find Mofit theme a perfect match. Also brings up some Trusted Solaris nostalgia :)
